pkg/services/authn/clients/api_key.go

```diff
@@ -62,6 +63,12 @@ func (s *APIKey) Authenticate(ctx context.Context, r *authn.Request) (*authn.Ide
 		return nil, errAPIKeyRevoked.Errorf("Api key is revoked")
 	}
 
+	if r.OrgID == 0 {
+		r.OrgID = apiKey.OrgID
```

Comment:
Should we reject `r.OrgID == 0`? Or are there valid scenarios where the global org ID would be used with an API key?

Reply:
`r.OrgID` is taken from either `X-Grafana-Org-Id` header or `targetOrgId` query string is present.

Before refactor the org id for the request was always set to the org the api key belongs to.
https://github.com/grafana/grafana/blob/v9.5.14/pkg/services/contexthandler/contexthandler.go#L407

Org id will be set to zero if neither the header or query string is present. So if we do a hard validation on org id 0 I think we would break a lot of stuff

Reply:
Thanks for explaining @kalleep and it's good to know this 👌